kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -0,0 +1,1206 @@
+
+      assertColumnMapping(updatedSchema.get("a"), 1)
+      assertColumnMapping(updatedSchema.get("array_of_arrays"), 4L, "array_of_arrays")
+
+      val arrayMetadata = updatedSchema.get("array_of_arrays").getMetadata
+      assert(arrayMetadata.getMetadata(ColumnMapping.COLUMN_MAPPING_NESTED_IDS_KEY)
+        .getLong("array_of_arrays.element") == 2L)
+      assert(arrayMetadata.getMetadata(ColumnMapping.COLUMN_MAPPING_NESTED_IDS_KEY)
+        .getLong("array_of_arrays.element.element") == 3L)
+    }
+  }
+
+  test("Changing column mapping on table and evolve schema at same time fails") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      val initialSchema = new StructType()
+        .add("a", StringType.STRING, true)
+        .add("c", IntegerType.INTEGER, true)
+
+      // Create a table initially without column mapping
+      createEmptyTable(engine, tablePath, initialSchema)
+
+      val currentSchema = table.getLatestSnapshot(engine).getSchema()
+      val newSchema = new StructType()
+        .add("a", StringType.STRING, true, currentSchema.get("a").getMetadata)
+        .add(
+          "b",
+          new StructType()
+            .add("d", IntegerType.INTEGER, true, fieldMetadataForColumn(4, "d"))
+            .add("e", IntegerType.INTEGER, true, fieldMetadataForColumn(5, "e")),
+          true,
+          fieldMetadataForColumn(3, "b"))
+        .add("c", IntegerType.INTEGER, true, currentSchema.get("c").getMetadata)
+
+      assertSchemaEvolutionFails[IllegalArgumentException](
+        table,
+        engine,
+        newSchema,
+        "Cannot update mapping mode and perform schema evolution",
+        Map(
+          TableConfig.COLUMN_MAPPING_MODE.getKey -> "name",
+          TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true"))
+    }
+  }
+
+  test("Updating schema on table when column mapping disabled fails") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      val initialSchema = new StructType()
+        .add("a", StringType.STRING, true)
+        .add("c", IntegerType.INTEGER, true)
+
+      createEmptyTable(engine, tablePath, initialSchema, tableProperties = Map.empty)
+
+      val currentSchema = table.getLatestSnapshot(engine).getSchema
+      val newSchema = new StructType()
+        .add("a", StringType.STRING, true, currentSchema.get("a").getMetadata)
+        .add(
+          "b",
+          new StructType()
+            .add("d", IntegerType.INTEGER, true, fieldMetadataForColumn(4, "d")),
+          true,
+          fieldMetadataForColumn(3, "b"))
+        .add("c", IntegerType.INTEGER, true, currentSchema.get("c").getMetadata)
+
+      assertSchemaEvolutionFails[KernelException](
+        table,
+        engine,
+        newSchema,
+        "Cannot update schema for table when column mapping is disabled")
+    }
+  }
+
+  test("Move a partition column") {
```

Comment:
Can we have a test that moves multiple partition columns? and changes their order?

i.e. `part_1, part_2, data_1` --> `part_2, part_1, data_1` ?

Btw -- if you used `part_X` in your test code -- it might make the code a bit easier to immediately discern which columns are in fact partition columns being tested/moved

Reply:
Oh yeah this is my bad naming...I'll fix this

Reply:
Fixed!